src/s2client-fsharp/ErrorDefs.fs

```diff
@@ -0,0 +1,16 @@
+﻿namespace Starcraft2
+
+type ApplicationError =
+    |FailedToEstablishConnection of string
```

Comment:
I'm okish with railway style as long as we properly forward error information (or logging it for that matter) instead of throwing everything away and only keeping the "Message" here.

"Standard" Exception handling has lots of features you now basically need to replicate manually.

Reply:
Agreed.